### PR TITLE
Update typescript to ^4.9.5 and and ttypescript to 1.5.15

### PR DESCRIPTION
**Pull request by cliffoo**
In trying to update typescript for all packages to 5.x I found out that the latest ttypescript [does not support typescript 5.x](https://github.com/cevek/ttypescript/issues/140), and it appears to be out of maintenance. We should consider forgoing the use of this package, or at least swapping it for an equivalent.

Here are the changes:
- Updated ttypescript to latest for all packages that use it (codec, compile-solidity, compile-solidity-tests, db, db-kit, expect, from-hardhat, hdwallet), so that they can use the latest typescript 4.x, which I then updated them to.
- Updated remaining packages to typescript 5.x.

Basically all ts packages use 5.x (^5.0.4) if they can, 4.x (^4.9.5) otherwise. Hopefully ci passes.

---

**Change of plans**: Only updating all to the latest 4.x ts, and we'll deal with ttypescript and updating ts to 5.x in the future. Still updating ttypescript.

### CODE REVIEW

**Comment by haltman-at** [comment]
Huh, do we want to be using multiple versions of typescript like that?

**Comment by cliffoo** [comment]
Ideally no but we currently have 4 versions 🤨 this cuts it to 2. I'm thinking packages stuck on 4.x can be transitioned over time to move on from ttypescript, and all catch up to using typescript 5.x at some point.

**Comment by haltman-at** [comment]
Oh, wow!  I thought we'd unified it some time ago but I guess not...

**Comment by haltman-at** [comment]
Hm, so looks like most things were on ^4.7.4, but dashboard-hardhat-plugin was on ^4.0.3, and from-hardhat was on ^4.3.5.  Wait, that's only three -- did I miss one?

**Comment by cliffoo** [comment]
I think preserve is locked at 4.7.4 without caret, lerna-update-wizard counted that as an additional version!

**Comment by eggplantzzz** [comment]
@gnidan can you weigh in on this??

**Review by gnidan — APPROVED** [review]
Good call to stick with ^4.9.5 until we can consider the ttypescript implications
